src/components/tree-item/tree-item.tsx

```diff
@@ -0,0 +1,291 @@
+import {
+  Component,
+  Event,
+  EventEmitter,
+  Element,
+  Prop,
+  State,
+  Listen,
+  h,
+  getAssetPath,
+  Watch
+} from "@stencil/core";
+@Component({
+  tag: "gxg-tree-item",
+  styleUrl: "tree-item.scss",
+  shadow: true,
+  assetsDirs: ["tree-item-assets"]
+})
+
+// return "./card-assets/new-card.svg";
+export class GxgTreeItem {
+  //PROPS
+  /**
+   * Set this attribute if you want the gxg-treeitem to display a checkbox
+   */
+  @Prop() checkbox = false;
+
+  /**
+   * Set this attribute if you want the gxg-treeitem checkbox to be checked by default
+   */
+  @Prop() checked = false;
+
+  /**
+   * Set this attribute when you are downloading a resource
+   */
+  @Prop() downloading = false;
```

Comment:
Is the purpose of this property to show feedback while the child nodes are loading?

Reply:
Yes, the purpose is to hide the +/- icon, and show the loader icon : 

```
  @Watch("downloading")
  downloadingHandler() {
    if (this.downloading) {
      this.hidePlusMinusIcon = true;
      setTimeout(() => (this.hidePlusMinusIcon = true), 500);
    } else {
      this.showDownloadingIcon = false;
      setTimeout(() => (this.hidePlusMinusIcon = false), 500);
    }
  }
```

Reply:
Yes. I listen to this property to hide/show the +/- icon and the loading icon as well.

```
  @Watch("downloading")
  downloadingHandler() {
    if (this.downloading) {
      this.hidePlusMinusIcon = true;
      setTimeout(() => (this.showDownloadingIcon = true), 250);
    } else {
      this.showDownloadingIcon = false;
      setTimeout(() => (this.hidePlusMinusIcon = false), 250);
    }
  }
```

check it here: https://www.awesomescreenshot.com/video/2607603?key=199f15b065e92411767ca0f69db3d609